x-pack/plugins/maps/common/descriptor_types/data_request_descriptor_types.ts

```diff
@@ -75,6 +75,7 @@ export type VectorStyleRequestMeta = MapFilters & {
 
 export type ESSearchSourceResponseMeta = {
   areResultsTrimmed?: boolean;
+  resultsCount?: number;
```

Comment:
Should this be optional? It should always be `resp.hits.hits.length` so `0` or more, or is that not right?

Reply:
Yes, because meta could be describing search hits or top hits. So resultsCount will not be present when using top hits.